src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java

```diff
@@ -119,7 +135,8 @@ public AWSCredentials getCredentials() {
                 initialCredentials = (new InstanceProfileCredentialsProvider()).getCredentials();
             }
 
-            AssumeRoleRequest assumeRequest = createAssumeRoleRequest(iamRoleArn);
+            AssumeRoleRequest assumeRequest = createAssumeRoleRequest(iamRoleArn)
+                    .withDurationSeconds(stsTokenDuration);
```

Comment:
The field can be `null`. Does this method allow a `null` parameter?

Reply:
I had thought that `defaultStsTokenDuration()` would ensure this is always populated with an integer. I might have been mistaken.
Would you recommend calling `getStsTokenDuration()` here instead?